src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -318,6 +332,22 @@ export class ExtHostTerminalService implements ExtHostTerminalServiceShape {
 		return renderer;
 	}
 
+	public async resolveTerminalRenderer(id: number): Promise<vscode.TerminalRenderer> {
+		// Check to see if the extension host already knows about this terminal.
+		for (const terminalRenderer of this._terminalRenderers) {
+			if (terminalRenderer._id === id) {
+				return terminalRenderer;
+			}
+		}
+
+		const dimensions = await this._proxy.$terminalGetDimensions(id);
```

Comment:
Can't this chunk just be this?

https://github.com/Microsoft/vscode/blob/1242997422e2fb4cb3e219b219be75d106fff8e9/src/vs/workbench/api/node/extHostTerminalService.ts#L315-L318

The idea behind the resolveTerminalRenderer idea was that you shouldn't need to change anything other than add this extra function which creates a renderer for an existing terminal? The task creates a Terminal which is owns completely, the extension host side can then create a TerminalRenderer for the Terminal the task system created.

Unless I'm missing something?

Reply:
I think there are conflicting requirements here. Originally I had this API take a vscode.terminal and a name (not sure why I name is needed since it's part of the terminal object), and did exactly what you are describing. @alexr00 asked me to change it to only take an ID.

So.... which way should we go?